src/hotspot/share/runtime/os.hpp

```diff
@@ -152,6 +152,25 @@ typedef void (*java_call_t)(JavaValue* value, const methodHandle& method, JavaCa
 
 class MallocTracker;
 
+// Preserve errno across a range of calls
+
+class ErrnoPreserver {
+  int _e;
+
+public:
+  ErrnoPreserver() {
+    _e = errno;
+  }
+
+  ~ErrnoPreserver() {
+    errno = _e;
+  }
+
+  int saved_errno() {
+    return _e;
+  }
```

Comment:
You could also make these functions one lines to save vertical white space, since they're short lines.